component/otelcol/processor/transform/transform_test.go

```diff
@@ -0,0 +1,549 @@
+package transform_test
+
+import (
+	"testing"
+
+	"github.com/grafana/agent/component/otelcol/processor/transform"
+	"github.com/grafana/river"
+	"github.com/mitchellh/mapstructure"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor"
+	"github.com/stretchr/testify/require"
+)
+
+func TestArguments_UnmarshalRiver(t *testing.T) {
+	tests := []struct {
+		testName string
+		cfg      string
+		expected map[string]interface{}
+		errorMsg string
+	}{
+		{
+			testName: "Defaults",
+			cfg: `
+			output {}
+			`,
+			expected: map[string]interface{}{
+				"error_mode": "propagate",
+			},
+		},
+		{
+			testName: "IgnoreErrors",
+			cfg: `
+			error_mode = "ignore"
+			output {}
+			`,
+			expected: map[string]interface{}{
+				"error_mode": "ignore",
+			},
+		},
+		{
+			testName: "TransformIfFieldDoesNotExist",
+			cfg: `
+			error_mode = "ignore"
+			trace_statements {
+				context = "span"
+				statements = [
+					// Accessing a map with a key that does not exist will return nil. 
+					"set(attributes[\"test\"], \"pass\") where attributes[\"test\"] == nil",
```

Comment:
Just a side-note: escaping the quotes seems to be quite painful at times. I think there was a plan to implement backticks "`" in River? We may want to update examples and tests after that's done.

Reply:
Yes! I see that there isn't yet an issue for raw strings, so I created one just now and mentioned this process in it:
https://github.com/grafana/river/issues/24